runtime/tchannel_client.go

```diff
@@ -193,6 +193,10 @@ func (c *TChannelClient) call(
 	if c.routingKey != nil {
 		ctxBuilder.SetRoutingKey(*c.routingKey)
 	}
+	rd, ok := ctx.Value("rd").(string)
+	if ok {
+		ctxBuilder.SetRoutingDelegate(rd)
```

Comment:
It would probably be a bit cleaner if instead of passing this parameter through the context it was passed as an optional parameter, e.g. [`yarpc.CallOption`](https://godoc.org/go.uber.org/yarpc#CallOption)

Reply:
That's not something easy to do at this moment, client call signature is `someMethod(ctx context.Context, reqHeaders map[string]string, args *SomeArgType)`.